src/CR.WorkingDayService/WorkingDayService.cs

```diff
@@ -0,0 +1,102 @@
+    {
+        private readonly IReadOnlyCollection<IWorkingDaySource> _sources;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WorkingDayService"/> class.
+        /// </summary>
+        /// <param name="sources">The <see cref="IWorkingDaySource"/>s to use to determine whether a given <see cref="DateTime"/> is on a working day, or a non-working day.
+        /// If no sources are specified, all days are considered working days.</param>
+        public WorkingDayService(IReadOnlyCollection<IWorkingDaySource> sources) => _sources = sources;
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Uses the <see cref="WorkingDayService"/>'s registered <see cref="IWorkingDaySource"/>s to determine if a particular <see cref="DateTime"/> is on a working day.
+        /// If there are no sources registered, all days are considered working days.
+        /// </summary>
+        public bool IsWorkingDay(DateTime date) => _sources == null || _sources.Count == 0 || _sources.Any(source => source.IsWorkingDay(date));
+
+        /// <summary>
+        /// Gets the next working day after the provided <see cref="DateTime"/>.
+        /// </summary>
+        /// <param name="date">The <see cref="DateTime"/> after which the soonest working day should be identified.</param>
+        /// <returns>The <see cref="DateTime"/> of the next working day after the provided date.</returns>
+        public DateTime NextWorkingDay(DateTime date)
+        {
+            do
+            {
+                date = date.AddDays(1);
+            }
+            while (this.IsNonWorkingDay(date));
+            return date;
+        }
+
+        /// <summary>
+        /// Gets the most recent working day from before the provided <see cref="DateTime"/>.
+        /// </summary>
+        /// <param name="date">The <see cref="DateTime"/> before which the most recent working day should be identified.</param>
+        /// <returns>The <see cref="DateTime"/> of the most recent working day before the provided date.</returns>
+        public DateTime PreviousWorkingDay(DateTime date)
+        {
+            do
+            {
+                date = date.AddDays(-1);
+            }
+            while (this.IsNonWorkingDay(date));
+            return date;
+        }
+
+        /// <summary>
+        /// Gets the date of the day on the provided number of working days after the provided <see cref="DateTime"/>.
+        /// </summary>
+        /// <param name="numberToAdd">The number of working days to add to the provided date.</param>
+        /// <param name="date">The <see cref="DateTime"/> to which working days should be added.</param>
+        /// <returns>The <see cref="DateTime"/> of the day on the provided number of working days after the provided date.</returns>
+        public DateTime AddWorkingDays(uint numberToAdd, DateTime date)
+        {
+            for (var i = 0; i < numberToAdd; i++)
+            {
+                do
```

Comment:
This is the same logic in `NextWorkingDay` and `PreviousWorkingDay`, we could implement `AddWorkingDays` and `SubtractWorkingDays` by calling the Next and Previous methods the correct number of times.